### PR TITLE
Refactor CancellationForm to be reusable

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Refactor cancellation form to be easier to reuse

## Justify why they are needed

- Preparing to implement it in the widget and offer presenter

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3795?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3787** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3787?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3795** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3795?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #3796** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3796?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3795?utm_source=stack-comment).

**Review by gustaveen — APPROVED** [review]
Oh, great initiative! 🙏

File: apps/store/src/components/Cancellation/CancellationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const [updateStartDate] = useStartDateUpdateMutation()
+  const handleChangeStartDate = (date: Date) => {
+    updateStartDate({
+      variables: {
+        productOfferIds,
+        startDate: formatAPIDate(date),
+      },
+    })
+  }
+
```
Comment (guilhermespopolin) [inline apps/store/src/components/Cancellation/CancellationForm.tsx:37]: Question: I saw in `SwitchPage` that we deal with [only one offer](https://github.com/HedvigInsurance/racoon/pull/3795/files#diff-8db0f50cd06f1a4265d968d9c662f306f44592d48785e501466e3520a13b5269L36):

```
const offer = props.priceIntent.offers[0]
```

Which is then passed in to `CancellationForm` as the `offer` prop. Wouldn't be the case of changing `handleAutoSwitchChange` and `handleChangeStartDate` functions so they only mutation `props.offer.id`?

```suggestion

  const [mutate] = useCancellationRequestedUpdateMutation()
  const handleAutoSwitchChange = (checked: boolean) => {
    mutate({ variables: { productOfferIds: [props.offer.id], requested: checked } })
  }

  const startDate = convertToDate(props.offer.startDate)
  const [updateStartDate] = useStartDateUpdateMutation()
  const handleChangeStartDate = (date: Date) => {
    updateStartDate({
      variables: {
        productOfferIds: [props.offer.id],
        startDate: formatAPIDate(date),
      },
    })
  }

```

**Review by guilhermespopolin — APPROVED** [review]
Nice refactoring :)
Just left a question

File: apps/store/src/components/Cancellation/CancellationForm.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/Cancellation/CancellationForm.tsx:37]: good point, however, we want to target all offers (tiers) when updating any of these properties

perhaps we should not always take the first offer from the price intent though

**Comment by robinandeer** [comment]
### Merge activity

* **Jan 3, 2:28 PM**: @robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3795).

**Pull request merged**